test/quicapitest.c

```diff
@@ -1553,6 +1554,521 @@ static int test_noisy_dgram(int idx)
+     * extension which was removed.
+     */
+    if (!TEST_true(qtest_fault_delete_extension(qtf, TLSEXT_TYPE_quic_transport_parameters,
+                                                ee->extensions, &ee->extensionslen,
+                                                old_bufm)))
+        goto err;
+
+    if (!TEST_true(PACKET_buf_init(&pkt, (unsigned char *)old_bufm->data, old_bufm->length))
+        || !TEST_ptr(new_bufm = BUF_MEM_new())
+        || !TEST_true(WPACKET_init(&wpkt, new_bufm)))
+        goto err;
+
+    have_wpkt = 1;
+
+    /*
+     * Open transport parameters TLS extension:
+     *
+     *   u16  Extension ID (quic_transport_parameters)
+     *   u16  Extension Data Length
+     *   ...  Extension Data
+     *
+     */
+    if (!TEST_true(WPACKET_put_bytes_u16(&wpkt,
+                                         TLSEXT_TYPE_quic_transport_parameters))
+        || !TEST_true(WPACKET_start_sub_packet_u16(&wpkt)))
+        goto err;
+
+    for (; PACKET_remaining(&pkt) > 0; ) {
+        tp_p = (unsigned char *)ossl_quic_wire_decode_transport_param_bytes(&pkt,
+                                                                            &id,
+                                                                            &tp_len);
+        if (!TEST_ptr(tp_p)) {
+            TEST_mem_eq(PACKET_data(&pkt), PACKET_remaining(&pkt), NULL, 0);
+            goto err;
+        }
+
+        if (!TEST_true(tparam_handle(ctx, id, tp_p, tp_len, &wpkt)))
+            goto err;
+    }
+
+    if (ctx->t->op == TPARAM_OP_INJECT || ctx->t->op == TPARAM_OP_DROP_INJECT
+        || ctx->t->op == TPARAM_OP_INJECT_TWICE) {
+        if (!TEST_ptr(ossl_quic_wire_encode_transport_param_bytes(&wpkt, ctx->t->id,
+                                                                  ctx->t->buf,
+                                                                  ctx->t->buf_len)))
+            goto err;
+
+        if (ctx->t->op == TPARAM_OP_INJECT_TWICE
+            && !TEST_ptr(ossl_quic_wire_encode_transport_param_bytes(&wpkt, ctx->t->id,
+                                                                     ctx->t->buf,
+                                                                     ctx->t->buf_len)))
+            goto err;
+    } else if (ctx->t->op == TPARAM_OP_INJECT_RAW) {
+        if (!TEST_true(WPACKET_memcpy(&wpkt, ctx->t->buf, ctx->t->buf_len)))
+            goto err;
+    }
+
+    if (!TEST_true(WPACKET_close(&wpkt))) /* end extension data, set length */
+        goto err;
+
+    if (!TEST_true(WPACKET_get_total_written(&wpkt, &written)))
+        goto err;
+
+    WPACKET_finish(&wpkt);
+    have_wpkt = 0;
+
+    /*
+     * Append the constructed extension blob to the extension block.
+     */
+    old_len = ee->extensionslen;
+
+    if (!qtest_fault_resize_message(qtf, ee->extensionslen + written))
+        goto err;
+
+    memcpy(ee->extensions + old_len, new_bufm->data, written);
```

Comment:
Side note: we maybe should have a `qtest_fault_add_extension`. Ok for this for now - but I wouldn't want to repeat this code in other tests.

Reply:
Yeah, it's definitely a gap.